Add dark theme for documentation

shibbirweb opened the pull request:
It will be very helpful for developer who likes dark theme for documentation.

taylorotwell commented:
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

shibbirweb commented:
@taylorotwell , Thank you Sir for your reply.

I'm describing what I added in this pull request

I just added created new file `_dark.scss` and update theme scheme of documentation.

and add those lines:

in `_color.scss`
```
// dark mode
$dark_primary_background: #242424;
$dark_secondary_background: #1b1b1b;
$dark_text: #e0e4e8;
$dark_text_secondary: #eeeeee;
```

in `_docs.scss`

```
.version_drop {
    ....
    margin-right: 2.25em;
}

.theme_mode {
 width: inherit;
}
```

in `app.scss`

```
@import "dark";
```

in `docs.blade.php`
```
                    <div class="theme_mode">
                        <div class="input-group">
                            <label>Docs Theme</label>
                            <div class="custom_select">
                                <select id="theme_switcher" aria-label="Laravel docs theme">
                                    <option value="light">Light</option>
                                    <option value="dark"
                                        {{ isset($_COOKIE['docs_theme']) && $_COOKIE['docs_theme'] == 'dark' ? 'selected' : '' }}>
                                        Dark</option>
                                </select>
                            </div>
                        </div>
                    </div>
```

in `layout.blade.php`
```
<body class="language-php {{ isset($_COOKIE['docs_theme']) && $_COOKIE['docs_theme'] == 'dark' ? 'dark' : '' }}">
```

in `docs.js`
```
$("#theme_switcher").change(function(e) {
    const theme = e.target.value;
    const path = "/docs";
    document.cookie = "docs_theme=" + theme + "; path=" + path;
    const body = $("body");
    if (theme === "light") {
        body.removeClass("dark");
    } else {
        body.addClass("dark");
    }
});
```

and `app.css` and `app.js` are the complied by npm to run in localhost. 

But rest changes in `layout.blade.php`, `docs.blade.php`, `app.scss`, `_prism.scss`, `_docs.scss`, `_color.scss` are changed by visual studio code extension [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) for indent correction.


I hope this documentation theme will be very helpful for work at nigh.



The previews are attached below

![Preview - Installation - Laravel - The PHP Framework For Web Artisans - localhost](https://user-images.githubusercontent.com/35910749/90105467-c2ee7980-dd67-11ea-80f7-2e096f34f3aa.png)

____
![Preview - Eloquent_ Relationships - Laravel - The PHP Framework For Web Artisan_ - localhost](https://user-images.githubusercontent.com/35910749/90106194-e4039a00-dd68-11ea-8b39-8fec3cf8b802.png)

WoLfulus commented:
I was going to make the same exact pull request, but seems like I'll just use Stylus to make it dark for now

Won't get accepted anyways :(

shibbirweb commented:
@WoLfulus I already created stylish theme for Laravel Dark Documentation.
Here is the link 
[Laravel Dark Documentation](https://userstyles.org/styles/187978/laravel-dark-documentation)
